lib/s3middleware/lifecycleHelpers/LifecycleUtils.ts

```diff
@@ -98,6 +139,42 @@ export default class LifecycleUtils {
         });
     }
 
+    /**
+    * Find the most relevant non-current version trantition rule for the given transitions array
+    * and any previously stored non-current version transition from another rule.
+    * @param params - The function parameters
+    * @param params.transitions - Array of lifecycle non-current version transitions
+    * @param params.lastModified - The object's last modified
+    * date
+    * @return The most applicable non-current version transition rule
+    */
+    getApplicableNCVTransition(params: {
```

Comment:
More of a question: given the new constraint that we can't transition out of cold storage (yet?) is it something that should be handled here, or at the call site? Call site seems to make more sense as the backbeat bucket processor has access to the location's `isCold` attribute.

Reply:
Yes, I think "backbeat bucket processor" is a good place to handle it. Also, we can't replication an object in a Cold storage location. We will need to handle this limitation as well. 

Note: We already handle replicating "to" a Cold storage location but not "from" there.

Reply:
I created a ticket for it: https://scality.atlassian.net/browse/BB-241